sklearn/linear_model/cd_fast.pyx

```diff
@@ -232,23 +232,20 @@ def enet_coordinate_descent(floating[::1] w,
                 # tmp = (X[:,ii]*R).sum()
                 tmp = dot(n_samples, &X[0, ii], 1, &R[0], 1)
 
-                if positive and tmp < 0:
+                if (positive or -alpha <= tmp) and (tmp <= alpha):
```

Comment:
to keep the idea of this check I would just write

`if positive and tmp < alpha:`

otherwise it's quite redundant with the else code block.

Reply:
Not sure I follow. This condition catches all cases where `w[ii]` would be zero to avoid the computation in the `else` block.

Reply:
I would benchmark all of these? Otherwise no way to know if any of them are faster than the current code, right?